### ISSUE TITLE
Option max_downloads not working on embed yt-dlp

### ISSUE THREAD

**Issue by Dysta** [yt-dlp/yt-dlp#2902]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

When embeding yt-dlp in my program, I try to limit the number of downloads from a playlist by using the `max_downloads` option.

On a shell, the feature work pretty well but when I'm using it on my Python code, the playlist is fully downloaded instead of the X elements as expected. 

See a reproducer:
```py
from yt_dlp import YoutubeDL

def main():
    YTDL_FORMAT_OPTION: dict = {
        "format": "bestaudio/best",
        'max_downloads': 5,
        "compat_opts": "no-youtube-unavailable-videos",
        "verbose": True,
    }
    with YoutubeDL(params=YTDL_FORMAT_OPTION) as ytdl:
        info = ytdl.extract_info(url="https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w", download=False, process=False)
        
    print(len(list(info["entries"])))  # 86
    

if __name__ == "__main__":
    main()
```

`len` should display 5 instead of 86. Also, I checked that the option exist and can be passed in the `params` dict and we're supposed to can (see here: https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/YoutubeDL.py#L3207)

Maybe i'm doing something wrong or idk

### Verbose log

```shell
[debug] Encodings: locale cp1252, fs utf-8, out cp1252 (No ANSI), err cp1252 (No ANSI), pref cp1252
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Compatibility options: n, o, -, y, o, u, t, u, b, e, -, u, n, a, v, a, i, l, a, b, l, e, -, v, i, d, e, o, s
[debug] Python version 3.9.9 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] exe versions: ffmpeg 4.4-essentials_build-www.gyan.dev (setts), ffprobe 4.4-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
[youtube:tab] PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w: Downloading webpage
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
[debug] [youtube:tab] Final URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
```

**Comment by Lesmiscore** [comment]
`max_downloads` is to limit number of downloads **when you use `download()` function**
https://github.com/yt-dlp/yt-dlp/blob/03f830040ae92af369ee046b082b1683ddf1539f/yt_dlp/YoutubeDL.py#L3207-L3209

I see you want to limit **number of items in playlists**. To do this, either use `'playlistend': 5` (equivalent to `--playlist-end 5`) or `'playlist_items': '1-5'` (`--playlist-items 1-5`)

**Comment by Dysta** [comment]
Thank you for your answer but playlistend seem to don't stop the download of a playlist to X elements
```py
from yt_dlp import YoutubeDL

def main():
    YTDL_FORMAT_OPTION: dict = {
        "format": "bestaudio/best",
        'playlistend': 5,
        "compat_opts": ["no-youtube-unavailable-videos"],
        "verbose": True,
    }
    with YoutubeDL(params=YTDL_FORMAT_OPTION) as ytdl:
        info = ytdl.extract_info(url="https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w", download=False, process=False)
        
    print(f"{len(list(info['entries']))=}")
    

if __name__ == "__main__":
    main()
```

```
[debug] Encodings: locale cp1252, fs utf-8, out cp1252 (No ANSI), err cp1252 (No ANSI), pref cp1252
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Compatibility options: no-youtube-unavailable-videos
[debug] Python version 3.9.9 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] exe versions: ffmpeg 4.4-essentials_build-www.gyan.dev (setts), ffprobe 4.4-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
[youtube:tab] PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w: Downloading webpage
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
[debug] [youtube:tab] Final URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
len(list(info['entries']))=86
```

**Comment by Lesmiscore** [comment]
> `process=False`

You've manually skipped processing!

**Comment by Dysta** [comment]
This isn't possible to skip de process and put a limit?

**Comment by Lesmiscore** [comment]
You can skip extraction of inner items using `'extract_flat': True` (`--flat-playlist`).

```python
from yt_dlp import YoutubeDL

def main():
    YTDL_FORMAT_OPTION: dict = {
        "format": "bestaudio/best",
        'playlistend': 5,
        "compat_opts": ["no-youtube-unavailable-videos"],
        "verbose": True,
        'extract_flat': True,
    }

    with YoutubeDL(params=YTDL_FORMAT_OPTION) as ytdl:
        info = ytdl.extract_info(url="https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w", download=False)
        
    print(f"{len(list(info['entries']))=}")
    

if __name__ == "__main__":
    main()
```

```
$ python3 a.py
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] ytdl-patched version 2022.02.04 
[debug] ** The command you are running is not yt-dlp.
[debug] ** Please make bug reports at https://github.com/ytdl-patched/ytdl-patched/issues/new/choose instead.
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Compatibility options: no-youtube-unavailable-videos
[debug] Python version 3.8.10 (CPython 64bit) - Linux-5.13.0-28-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4, rtmpdump 2.4
[debug] Optional libraries: secretstorage, sqlite
[debug] Proxy map: {}
[debug] [youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
[youtube:tab] PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w: Downloading webpage
[youtube:tab] [debug] Fetching webpage from https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
[debug] [youtube:tab] Final URL: https://www.youtube.com/playlist?list=PLF_ZnpSKNQQFTeHaQ1ZR5l5cx5nRiD80w
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
[download] Downloading playlist: 3AM (slowed + reverb)
[youtube:tab] playlist 3AM (slowed + reverb): Downloading 5 videos
[download] Downloading video 1 of 5
[download] Downloading video 2 of 5
[download] Downloading video 3 of 5
[download] Downloading video 4 of 5
[download] Downloading video 5 of 5
[download] Finished downloading playlist: 3AM (slowed + reverb)
len(list(info['entries']))=5
```

**Comment by Dysta** [comment]
Yes, it's what I need
